resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -187,6 +189,10 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val schedulerUuid: String = UUID.randomUUID().toString
   private val nextExecutorNumber = new AtomicLong()
 
+  private val mesosReviveThread =
+    ThreadUtils.newDaemonSingleThreadScheduledExecutor("mesos-revive-thread")
+  private val mesosReviveIntervalMs = conf.getTimeAsMs("spark.scheduler.revive.interval", "1000ms")
```

Comment:
1s seems too aggressive. Since this is mainly to guard against the edge case of revive being dropped, this could be changed to something higher, maybe 10s. Also, I would suggest to use a different new config option than piggybacking on the  `spark.scheduler.revive.interval ` which seems to be for a different thing?

Reply:
Changed to a new config option `spark.scheduler.revive.interval`. Default - 10 seconds.